src/main/java/com/jmb/springfactory/dao/GenericMongoDao.java

```diff
@@ -0,0 +1,38 @@
+package com.jmb.springfactory.dao;
+
+import java.io.Serializable;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import com.jmb.springfactory.exceptions.PersistenceLayerException;
+import com.jmb.springfactory.model.entity.BaseEntity;
+
+public interface GenericMongoDao<T extends BaseEntity, ID extends Serializable> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00119)